library/src/main/java/com/google/android/exoplayer/metadata/Id3Parser.java

```diff
@@ -67,6 +67,39 @@ public boolean canParse(String mimeType) {
         String value = new String(frame, valueStartIndex, valueEndIndex - valueStartIndex,
             charset);
         metadata.put(TxxxMetadata.TYPE, new TxxxMetadata(description, value));
+      } else if (frameId0 == 'P' && frameId1 == 'R' && frameId2 == 'I' && frameId3 == 'V') {
+        // Check frame ID == PRIV
+        byte[] frame = new byte[frameSize];
+        id3Data.readBytes(frame, 0, frameSize);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String owner = new String(frame, 0, firstZeroIndex);
+        byte[] privateData = new byte[frameSize - firstZeroIndex - 1];
+        System.arraycopy(frame, firstZeroIndex + 1, privateData, 0, frameSize - firstZeroIndex - 1);
+        metadata.put(PrivMetadata.TYPE, new PrivMetadata(owner, privateData));
+      } else if (frameId0 == 'G' && frameId1 == 'E' && frameId2 == 'O' && frameId3 == 'B') {
+        // Check frame ID == GEOB
+        int encoding = id3Data.readUnsignedByte();
+        String charset = getCharsetName(encoding);
+        byte[] frame = new byte[frameSize - 1];
+        id3Data.readBytes(frame, 0, frameSize - 1);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String mimeType = new String(frame, 0, firstZeroIndex);
+        int filenameStartIndex = firstZeroIndex + 1;
+        int filenameEndIndex = indexOf(frame, filenameStartIndex, (byte) 0);
+        String filename = new String(frame, filenameStartIndex,
+                                     filenameEndIndex - filenameStartIndex, charset);
+        int descriptionStartIndex = filenameEndIndex + 1;
```

Comment:
Use indexOfNot(frame, filenameEndIndex, (byte) 0); In some encodings it could be a single 0 byte delimiter, in some it might be two 0 bytes. Same applies for the objectData start. I'd also use it for filenameStartIndex for consistency.